charts/argo-cd/templates/dex/certificate.yaml

```diff
@@ -0,0 +1,38 @@
+{{- if and .Values.dex.enabled .Values.dex.certificate.enabled -}}
+apiVersion: {{ include "argo-cd.apiVersion.cert-manager" . }}
+kind: Certificate
+metadata:
+  name: {{ include "argo-cd.dex.fullname" . }}
+  labels:
+    {{- include "argo-cd.labels" (dict "context" . "component" .Values.dex.name "name" .Values.dex.name) | nindent 4 }}
+spec:
+  secretName: argocd-dex-server-tls
+  {{- with .Values.dex.certificate.domain }}
+  commonName: {{ . | quote }}
```

Comment:
Common name is intentionally kept for external PKI providers that require this to be set as it might be convenient to have different certs for multiple Argo / Dex instances.